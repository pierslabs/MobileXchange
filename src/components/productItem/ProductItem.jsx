import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ProductItem = ({ product }) => {
  return (
    <Link
      to={`/products/product/${product.id}`}
      className='flex flex-col justify-center items-center bg-white p-4 shadow-md rounded-lg transition-transform transform-gpu hover:scale-105 '
    >
      <img src={product.imgUrl} alt={product.model} />
      <h3 className='text-lg font-semibold'>{product.brand}</h3>
      <p className='text-gray-600'>{product.model}</p>
      <p className='text-gray-600'>{product.price}</p>
    </Link>
  );
};

export default ProductItem;

ProductItem.propTypes = {
  product: PropTypes.shape({
    brand: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
  }).isRequired,
};
