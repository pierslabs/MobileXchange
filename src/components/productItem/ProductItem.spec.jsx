import { describe, expect, it } from 'vitest';
import ProductItem from './ProductItem';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe('ProductItem', () => {
  it('should render correctly', () => {
    const product = {
      brand: 'Apple',
      id: '1',
      imgUrl: 'https://via.placeholder.com/150',
      model: 'iPhone 12',
      price: '799',
    };
    const sut = render(
      <MemoryRouter>
        <ProductItem product={product} />
      </MemoryRouter>
    );

    expect(sut).toMatchSnapshot();
  });
});
