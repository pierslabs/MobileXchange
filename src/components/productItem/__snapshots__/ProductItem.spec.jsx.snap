// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductItem > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="items-center border p-4 shadow-lg rounded-lg transition-transform  hover:scale-105 card overflow-hidden"
        href="/products/product/1"
      >
        <div
          class="flex justify-between items-center mb-3"
        >
          <h3
            class="sm:text-2xl  text-gray-700 sm:-rotate-45 text-center underline"
          >
            Apple
          </h3>
          <p
            class="p-1 text-sm sm:text-lg bg-yellow-400 transform -skew-x-12"
          >
            <span
              class=" line-through text-red-600  sm:mr-2"
            >
              958,80 €
            </span>
          </p>
        </div>
        <img
          alt="iPhone 12"
          class="mx-auto w-auto h-auto false "
          loading="lazy"
          src="https://via.placeholder.com/150"
        />
        <div
          class="sm:flex text-center text-sm items-center justify-between mt-10 flex-wrap w-full"
        >
          <h3
            class="sm:text-xl "
          >
            iPhone 12
          </h3>
          <p
            class="text-2xl text-orange-500 mt-2"
          >
            799,00 €
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="items-center border p-4 shadow-lg rounded-lg transition-transform  hover:scale-105 card overflow-hidden"
      href="/products/product/1"
    >
      <div
        class="flex justify-between items-center mb-3"
      >
        <h3
          class="sm:text-2xl  text-gray-700 sm:-rotate-45 text-center underline"
        >
          Apple
        </h3>
        <p
          class="p-1 text-sm sm:text-lg bg-yellow-400 transform -skew-x-12"
        >
          <span
            class=" line-through text-red-600  sm:mr-2"
          >
            958,80 €
          </span>
        </p>
      </div>
      <img
        alt="iPhone 12"
        class="mx-auto w-auto h-auto false "
        loading="lazy"
        src="https://via.placeholder.com/150"
      />
      <div
        class="sm:flex text-center text-sm items-center justify-between mt-10 flex-wrap w-full"
      >
        <h3
          class="sm:text-xl "
        >
          iPhone 12
        </h3>
        <p
          class="text-2xl text-orange-500 mt-2"
        >
          799,00 €
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
